### ISSUE TITLE
[Tabs] Show badge next to tab title

### ISSUE THREAD

**Issue by amireynoso** [Shopify/polaris-react#3162]
## Feature request summary
We'd like to be able to show a badge (in any state & with any content) next to the title of a tab.

**Screenshot**
<img width="553" alt="image" src="https://user-images.githubusercontent.com/1855829/89830148-c1377280-db29-11ea-8cc9-1443d5be2b38.png">

## Rationale
In our particular use case, we'd like to show a count of resources listed within a tab. However, I can see the use of a badge to also indicate other things (e.g. `New`, `Beta`, etc).

## Additional context
This is something we're looking to include in a new feature we're working on the Vault, so we're more than happy to take ownership of this feature (i.e. open PR and ensure it's released).

cc @KwanLam17 

---

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md) for more information.

**Comment by BPScott** [comment]
Go for it!

This looks like [it's mostly already possible](https://codesandbox.io/s/winter-cookies-phfoy?file=/App.js) from a very simplistic implementation perspective. I think it should just be a case of updating the types on [`TabDescriptor['content']`](https://github.com/Shopify/polaris-react/blob/master/src/components/Tabs/types.ts#L7) from `string` to `React.ReactNode`. https://github.com/Shopify/polaris-react/pull/2972 is a prior example of type loosening.

You should also add an example to the Tabs readme showcasing this usage.


---

Previously we've been tried to be somewhat restrictive around these APIs (e.g. Navigation.Item has a badge prop) but that convention doesn't really stop mis-use as in JS land you can ignore the type warnings and people always end up finding a legit usage for ReactNode anyway so I'm thinking we skip that intermediate step of trying to be strict just go straight to the endgame.

**Comment by amireynoso** [comment]
Hey @BPScott, thanks for the context, that's an interesting approach! 

@KwanLam17 and I took a stab at it following the implementation on `<ActionList />` (not documented) and it was pretty straight-forward. Basically, what we did was add a `badge` prop (as defined in `BadgeAction`) to `TabDescriptor`, and use it to display a Badge component.

But your approach is much simpler and flexible. So just to confirm, are we cool with opening this value up to be any React node?

Let me know, we'll open a PR shortly. 🙏
